src/pal/src/pal_init_read.c

```diff
@@ -0,0 +1,14 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+char *read_resolv_conf(void) {
+    FILE *fp = fopen("/etc/resolv.conf", "rb");
+    fseek(fp, 0, SEEK_END);
+    long fsize = ftell(fp);
+    fseek(fp, 0, SEEK_SET);
+    char *resolv_conf_buffer = malloc(fsize + 1);
```

Comment:
Memory leak, no-one would free this buffer.

Reply:
Done